mlir/include/mlir/Dialect/Ptr/IR/PtrOps.td

```diff
@@ -52,6 +121,36 @@ def Ptr_PtrAddOp : Pointer_Op<"ptr_add", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// ToPtrOp
+//===----------------------------------------------------------------------===//
+
+def Ptr_ToPtrOp : Pointer_Op<"to_ptr", [Pure]> {
```

Comment:
Can this be a replacement for `memref.extract_aligned_pointer_as_index`? Do we have ways to "cast" a `!ptr.ptr` to an `index`/`i64`/...?

Reply:
This is one 100% a replacement. `to_ptr %memref` should give you a ptr to the aligned buffer, and already as a ptr type instead of an it.

Reply:
How could existing IR such as the following one be rewritten with `to_ptr` instead of `extract_aligned_pointer_as_index`?

```mlir
%0 = memref.extract_aligned_pointer_as_index %m : memref<10xf32>
vector.print %0 : index
```